packages/web-components/tests/e2e/build/build-examples.js

```diff
@@ -0,0 +1,239 @@
+  .option('-P, --skip-packages', 'Skips the local package setup')
+  .option('-E, --skip-examples', 'Skips the local examples setup');
+program.parse();
+
+/**
+ * Command line options
+ *
+ * @type {commander.OptionValues}
+ */
+const _opts = program.opts();
+
+const { log, error } = console;
+
+/**
+ * Project root folder
+ *
+ * @type {string}
+ * @private
+ */
+const _projectRoot = path.resolve(__dirname, '../../..');
+
+/**
+ * Test script folder
+ *
+ * @type {string}
+ * @private
+ */
+const _testScriptFolder = `${_projectRoot}/tests/e2e/build`;
+
+/**
+ * Examples source folder
+ *
+ * @type {string}
+ * @private
+ */
+const _exampleSrc = path.resolve(__dirname, '../../../examples/codesandbox/components');
+
+/**
+ * Local packages for the examples
+ *
+ * @type {string}
+ * @private
+ */
+const _localPackagesFolder = mkdirSync('temp-packages-');
+
+/**
+ * Examples build folder
+ *
+ * @type {string}
+ * @private
+ */
+const _exampleBuild = mkdirSync('temp-examples-');
+
+/**
+ * Distribution folder
+ *
+ * @type {string}
+ * @private
+ */
+const _distFolder = path.resolve(__dirname, '../dist/');
+
+/**
+ * Defines the local packs to build
+ *
+ * @type {{styles: string, services: string, utilities: string, 'web-components': string}}
+ * @private
+ */
+const _packages = {
+  services: path.resolve(__dirname, '../../../../services'),
+  styles: path.resolve(__dirname, '../../../../styles'),
+  utilities: path.resolve(__dirname, '../../../../utilities'),
+  'web-components': path.resolve(__dirname, '../../..'),
+};
+
+/**
+ * Stores the list of examples
+ *
+ * @type {*[]}
+ * @private
+ */
+let _examples = [];
+
+/**
+ * Gets the list of directories in the examples folder
+ *
+ * @param {string} folder Folder to get the list of sub directories from
+ * @returns {*} Array of folders
+ * @private
+ */
+function _getDirectories(folder) {
+  return fs
+    .readdirSync(folder, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .map(dirent => dirent.name);
+}
+
+/**
+ * Puts the build artifacts in the given temporary directory
+ * and resolves dependencies in `package.json` to use such build artifacts.
+ *
+ * @private
+ */
+function _setupPackages() {
+  if (!_opts.skipPackages) {
+    log(chalk.yellow('Creating local packages...'));
+
+    Object.keys(_packages).forEach(pack => {
+      log(chalk.green(`Building package: ${pack}`));
+
+      const commands = [];
+      commands.push(
+        `cd "${_packages[pack]}" && yarn pack --filename "${_localPackagesFolder}/carbon-ibmdotcom-${pack}.tar.gz"`,
+        `tar xzf "${_localPackagesFolder}/carbon-ibmdotcom-${pack}.tar.gz" --directory "${_localPackagesFolder}"`,
+        `mv "${_localPackagesFolder}/package" "${_localPackagesFolder}/ibmdotcom-${pack}"`,
+        // eslint-disable-next-line max-len
+        `node "${_testScriptFolder}/replace-dependencies.js" -f "${_localPackagesFolder}" "${_localPackagesFolder}/ibmdotcom-${pack}/package.json"`
```

Comment:
Just a comment but we can probably test this out later and see if skipping the decompression and linking the tarball directly into the `package.json` could maybe squeeze out a little bit of performance.

Reply:
sure, most of this was pre-existing and largely kept this part the same other than different paths.